featurebyte/worker/task/deployment_create_update.py

```diff
@@ -5,23 +5,43 @@
 
 from typing import Any, cast
 
+from featurebyte.exception import DocumentCreationError, DocumentUpdateError
 from featurebyte.schema.worker.task.deployment_create_update import (
     CreateDeploymentPayload,
     DeploymentCreateUpdateTaskPayload,
     DeploymentPayloadType,
     UpdateDeploymentPayload,
 )
-from featurebyte.worker.task.base import BaseTask
+from featurebyte.worker.task.base import BaseLockTask
 
 
-class DeploymentCreateUpdateTask(BaseTask):
+class DeploymentCreateUpdateTask(BaseLockTask):
     """
     FeatureList Deploy Task
     """
 
     payload_class = DeploymentCreateUpdateTaskPayload
 
-    async def execute(self) -> Any:
+    @property
+    def lock_key(self) -> str:
+        # each deployment can only be created or updated once at a time
+        payload = cast(DeploymentCreateUpdateTaskPayload, self.payload)
+        return f"deployment:{payload.output_document_id}:create_update"
+
+    @property
+    def lock_blocking(self) -> bool:
+        # if the deployment is being created or updated by another task,
+        # fail the current task immediately without waiting
+        return False
+
+    def handle_lock_not_acquired(self) -> None:
+        payload = cast(DeploymentCreateUpdateTaskPayload, self.payload)
+        error_msg = f"Deployment {payload.output_document_id} is currently being created or updated"
```

Comment:
nit: Maybe can customise the error message based on deployment payload type.

Reply:
We can't customize the error message based on the deployment payload type here unless we make another mongo call to retrieve the task payload (current type may not be the same as the one running on the other task).